src/Microsoft.Framework.PropertyActivator.Sources/PropertyActivator.cs

```diff
@@ -34,14 +34,31 @@ public object Activate([NotNull] object view, [NotNull] TContext context)
             [NotNull] Type activateAttributeType,
             [NotNull] Func<PropertyInfo, PropertyActivator<TContext>> createActivateInfo)
         {
-            return type.GetRuntimeProperties()
-                       .Where(property =>
-                              property.IsDefined(activateAttributeType) &&
-                              property.GetIndexParameters().Length == 0 &&
-                              property.SetMethod != null &&
-                              !property.SetMethod.IsStatic)
-                       .Select(createActivateInfo)
-                       .ToArray();
+            return GetPropertiesToActivate(type, activateAttributeType, createActivateInfo, includeNonPublic: false);
+        }
+
+        public static PropertyActivator<TContext>[] GetPropertiesToActivate(
+            [NotNull] Type type,
+            [NotNull] Type activateAttributeType,
+            [NotNull] Func<PropertyInfo, PropertyActivator<TContext>> createActivateInfo,
+            bool includeNonPublic)
+        {
+            var properties = type.GetRuntimeProperties()
+                .Where((property) =>
+                {
+                    return
+                        property.IsDefined(activateAttributeType) &&
+                        property.GetIndexParameters().Length == 0 &&
+                        property.SetMethod != null &&
+                        !property.SetMethod.IsStatic;
+                });
+
+            if (!includeNonPublic)
+            {
+                properties = properties.Where(property => property.SetMethod.IsPublic);
+            }
```

Comment:
Disagree. This "but wait there's more" approach is unreadable and wasted my time as I tried to figure out a solid reason why checking `includeNonPublic` and `property.SetMethod.IsPublic` wasn't done inside the previous lambda. If you don't like a parenthesized `||` expression, fine. The lambda statement block could contain
``` c#
  var result = {existing condition};
  if (result)
  {
    return includeNonPublic || property.SetMethod.IsPublic;
  }

  return false;
```
or (just to avoid a visible `||`)
``` c#
  ...
  if (result && includeNonPublic)
  {
    return true;
  }

  return result && property.SetMethod.IsPublic;
```